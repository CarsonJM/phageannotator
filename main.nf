#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    nf-core/phageannotator
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Github : https://github.com/nf-core/phageannotator
    Website: https://nf-co.re/phageannotator
    Slack  : https://nfcore.slack.com/channels/phageannotator
----------------------------------------------------------------------------------------
*/

nextflow.enable.dsl = 2

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { PHAGEANNOTATOR  } from './workflows/phageannotator'
include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_phageannotator_pipeline'
include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_phageannotator_pipeline'

include { getGenomeAttribute      } from './subworkflows/local/utils_nfcore_phageannotator_pipeline'

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    GENOME PARAMETER VALUES
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

// TODO nf-core: Remove this line if you don't need a FASTA file
//   This is an example of how to use getGenomeAttribute() to fetch parameters
//   from igenomes.config using `--genome`
params.fasta = getGenomeAttribute('fasta')

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    VALIDATE & PRINT PARAMETER SUMMARY
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { validateParameters; paramsHelp } from 'plugin/nf-validation'

// Print help message if needed
if (params.help) {
    def logo = NfcoreTemplate.logo(workflow, params.monochrome_logs)
    def citation = '\n' + WorkflowMain.citation(workflow) + '\n'
    def String command = "nextflow run ${workflow.manifest.name} --input samplesheet.csv --genome GRCh37 -profile docker"
    log.info logo + paramsHelp(command) + citation + NfcoreTemplate.dashedLine(params.monochrome_logs)
    System.exit(0)
}

// Validate input parameters
if (params.validate_params) {
    validateParameters()
}

WorkflowMain.initialise(workflow, params, log, args)

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    NAMED WORKFLOW FOR PIPELINE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// WORKFLOW: Run main analysis pipeline depending on type of input
//
workflow NFCORE_PHAGEANNOTATOR {

    take:
    samplesheet // channel: samplesheet read in from --input

    main:

    //
    // WORKFLOW: Run pipeline
    //
    PHAGEANNOTATOR (
        samplesheet
    )

    emit:
    multiqc_report = PHAGEANNOTATOR.out.multiqc_report // channel: /path/to/multiqc_report.html

}
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN MAIN WORKFLOW
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

workflow {
    NFCORE_PHAGEANNOTATOR ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
