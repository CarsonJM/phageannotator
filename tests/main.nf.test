nextflow_pipeline {

    name "Test pipeline: NFCORE_PHAGEANNOTATOR"
    script "main.nf"
    tag "pipeline"
    tag "nf-core-phageannotator"
    tag "nf-core-phageannotator_default"

    test("Parameters: default") {
        when {
            params {
                outdir  = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${params.outdir}/ReadQuality"),
                    path("${params.outdir}/AssemblyFilter"),
                    path("${params.outdir}/ReferenceContainment"),
                    path("${params.outdir}/VirusClassification"),
                    path("${params.outdir}/VirusQuality"),
                    path("${params.outdir}/pipeline_info/software_versions.yml")
                    ).match() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() }
            )
        }
    }
}
