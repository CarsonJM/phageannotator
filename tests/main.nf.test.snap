{
    "Parameters: default": {
        "content": [
            [
                [
                    "test1_1_fastqc.html:md5,78eccd656daaf169087de74e7190b810",
                    "test1_2_fastqc.html:md5,4615c73b62961e656852a2d3424dbe61"
                ]
            ],
            [
                [
                    [
                        "test1.fasta.gz:md5,b91bcdca503b2b02005cae7fc5a594b7"
                    ]
                ]
            ],
            [
                "all_samples.screen:md5,b074c6861bab21e2b3aa63aebff1f392",
                [
                    "test1.fasta_w_screen_hits.fna.gz:md5,a8a0353be8912b70ca8c3fb370f9134a"
                ],
                [
                    [
                        "test1.screen:md5,b074c6861bab21e2b3aa63aebff1f392"
                    ]
                ]
            ],
            [
                "all_samples.tsv:md5,8aa138f16ebdaf23163791d8bba3427a",
                [
                    [
                        [
                            "test1.fasta_w_screen_hits_virus.fna.gz:md5,b06f4bc2e21a17122fb999ba661ab821",
                            "test1.fasta_w_screen_hits_virus_summary.tsv:md5,8aa138f16ebdaf23163791d8bba3427a"
                        ]
                    ]
                ]
            ],
            [
                "all_samples.tsv:md5,c96fdb5e5f179205fe23cee7a194c270",
                [
                    [
                        [
                            "proviruses.fna.gz:md5,7029066c27ac6f5ef18d660d5741979a",
                            "quality_summary.tsv:md5,c96fdb5e5f179205fe23cee7a194c270",
                            "viruses.fna.gz:md5,82bde1db0b611b3f2a2921a00831db97"
                        ]
                    ]
                ],
                [
                    "test1.filtered.fna.gz:md5,77d66bcbe7149649de3d1c9e4bc9a1b1"
                ]
            ],
            "software_versions.yml:md5,72a22a1552bddbce6500a2aff2253065"
        ],
        "timestamp": "2023-10-03T09:55:44.763311842"
    }
}